tests/sentry/tasks/test_groupowner.py

```diff
@@ -0,0 +1,264 @@
+
+        self.set_release_commits(self.user_2.email)
+        process_suspect_commits(self.event_2.group_id, cache_key)
+        assert GroupOwner.objects.filter(group=self.event_1.group).count() == 2
+        assert GroupOwner.objects.filter(group=self.event_1.group, user=self.user).exists()
+        assert GroupOwner.objects.filter(group=self.event_2.group, user=self.user_2).exists()
+
+        data = self.event_3.data
+        data["event_id"] = self.event_3.event_id
+        data["project"] = self.event_3.project_id
+        cache_key = event_processing_store.store(data)
+        self.set_release_commits(self.user_3.email)
+        process_suspect_commits(self.event_3.group_id, cache_key)
+        assert GroupOwner.objects.filter(group=self.event_1.group).count() == 3
+        assert GroupOwner.objects.filter(group=self.event_1.group, user=self.user).exists()
+        assert GroupOwner.objects.filter(group=self.event_2.group, user=self.user_2).exists()
+        assert GroupOwner.objects.filter(group=self.event_2.group, user=self.user_3).exists()
+
+        data = self.event_4.data
+        data["event_id"] = self.event_4.event_id
+        data["project"] = self.event_4.project_id
+        cache_key = event_processing_store.store(data)
+        self.set_release_commits(self.user_4.email)
+        process_suspect_commits(self.event_4.group_id, cache_key)
+        assert GroupOwner.objects.filter(group=self.event_1.group).count() == 3
+        assert GroupOwner.objects.filter(group=self.event_1.group, user=self.user).exists()
+        assert GroupOwner.objects.filter(group=self.event_2.group, user=self.user_2).exists()
+        assert GroupOwner.objects.filter(group=self.event_2.group, user=self.user_3).exists()
+        assert not GroupOwner.objects.filter(group=self.event_2.group, user=self.user_4).exists()
+
+        go = GroupOwner.objects.filter(group=self.event_2.group, user=self.user_2).first()
+        go.date_added = timezone.now() - PREFERRED_GROUP_OWNER_AGE * 2
+        go.save()
+
+        data = self.event_4.data
+        data["event_id"] = self.event_4.event_id
+        data["project"] = self.event_4.project_id
+        cache_key = event_processing_store.store(data)
+        self.set_release_commits(self.user_4.email)
+        process_suspect_commits(self.event_4.group_id, cache_key)
+        assert GroupOwner.objects.filter(group=self.event_1.group).count() == 3
+        assert GroupOwner.objects.filter(group=self.event_1.group, user=self.user).exists()
+        assert not GroupOwner.objects.filter(group=self.event_2.group, user=self.user_2).exists()
+        assert GroupOwner.objects.filter(group=self.event_2.group, user=self.user_3).exists()
+        assert GroupOwner.objects.filter(group=self.event_2.group, user=self.user_4).exists()
+
+    def test_update_existing_entries(self):
+        # As new events come in associated with existing owners, we should update the date_added of that owner.
+        pass
```

Comment:
Should this be implemented?

Reply:
I was just going to do it in the test above, but I've gone ahead and implemented this